### PR TITLE
set up the server routes, hooked up mongoDB, created gulp commands

### PR DESCRIPTION
**Pull request by aamangeldi**
After pulling these changes make sure to:
- create a directory named "db" for the database files in the main directory,
- download mongodb (https://www.mongodb.com/)
- download gulpJS globally (npm install gulp-cli -g)
- run 'npm install' in the main directory

In the main directory, run "gulp start-fresh", which will start up the database (port 4201) with new data from the "test_data" folder, the server (port 4200), and the app itself (port 3000).

The routes are set up to GET, PUT, POST, and DELETE from/to the two collections in the database: tutorials and users. GET, PUT, and DELETE can work for individual documents in the collections, just add /Object_id to the route when testing.

### CODE REVIEW

**Comment by aamangeldi** [comment]
POSTMAN is a very helpful tool to test the Express routes. It works as a simple client, from which you can hit the routes.

https://www.getpostman.com/

File: app.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+  db = database;
+
+	// then start the server
+	server.listen(PORT);
+	console.log('Listening on port %d', server.address().port);
+});
+
+// setup the routes
+app.get('/', (req, res)=>{
+	res.send("<h1>Homepage</h1>");
```
Comment (jgnickerson) [inline app.js:46]: I'll hook this up to serve the webpack bundle tomorrow

File: app.js [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+});
+
+// get the info about the tutorials
+app.get('/tutorials/:id*?', (req, res) => {
```
Comment (jgnickerson) [inline app.js:50]: What do you think about using the mongo drivers ES6 features (promises, generators), seen here: https://mongodb.github.io/node-mongodb-native/2.2/tutorials/crud/
There's a little slider on the site to switch from Javascript 5 to JS6, so make sure you toggle that to see what I'm talking about.

I think it's cleaner, especially when there's multiple queries and error catching.

The "co" thing they are using is explained here: https://github.com/tj/co

File: app.js (hunk 1 above)
Comment (aamangeldi) [reply app.js:46]: Sounds good!

File: app.js (hunk 2 above)
Comment (aamangeldi) [reply app.js:50]: @jgnickerson sure let's try ES6 -- should we try it with this push? or should we just open a new issue to re-write this code later on?

Comment (jgnickerson) [reply app.js:50]: I think I'll merge this in and then we can rewrite it soon. Shouldn't take too long.

Comment (aamangeldi) [reply app.js:50]: @jgnickerson sounds good to me! let me know if you need help with this!

**Comment by jgnickerson** [comment]
LGTM (looks good to me). I think the ES6 stuff we can play with down the road.

**Pull request merged**